content/en/docs/how-tos/external-images.md

```diff
@@ -33,3 +33,23 @@ base_images:
     name:  boskos
     tag: latest
 {{< / highlight >}}
+
+Note that if your image is in a private registry that requires authentication to pull it, you will need to [add your credentials](/docs/how-tos/adding-a-new-secret-to-ci/) and ensure that you use the [`Local`](https://docs.openshift.com/container-platform/4.7/rest_api/image_apis/imagestream-image-openshift-io-v1.html) `referencePolicy` on your `ImageStream` to allow downstream consumers to not require authentication:
+
+{{< highlight yaml >}}
+apiVersion: image.openshift.io/v1
+kind: ImageStream
+metadata:
+  name: secret
+  namespace: my-namespace
+spec:
```

Comment:
```suggestion
spec:
  lookupPolicy:
    local: true
```

Otherwise it will still try to contact the registry to try to resolve the tag (but would then pull from cluster but it never gets there). At least I think that is how it works?

Reply:
No, `lookupPolicy` is used to allow an OpenShift admission webhook to change references like `myimage` to `<local-registry>/the-namespace/my-image:latest`:

```
)$ oc --context app.ci explain imagestream.spec.lookupPolicy
KIND:     ImageStream
VERSION:  image.openshift.io/v1

RESOURCE: lookupPolicy <Object>

DESCRIPTION:
     lookupPolicy controls how other resources reference images within this
     namespace.

     ImageLookupPolicy describes how an image stream can be used to override the
     image references used by pods, builds, and other resources in a namespace.

FIELDS:
   local	<boolean> -required-
     local will change the docker short image references (like "mysql" or
     "php:latest") on objects in this namespace to the image ID whenever they
     match this image stream, instead of reaching out to a remote registry. The
     name will be fully qualified to an image ID if found. The tag's
     referencePolicy is taken into account on the replaced value. Only works
     within the current namespace.
```

Reply:
I don't understand it, how can the kubelet if this is `false` tell if something like `mysql` is supposed to reference `docker.io/mysql:latest` or the imagestreamtag `mysql:latest`? (Happy to read docs about this but I am genuinely confused. I also remember we had to set this to true for the testimagestreamdistributioncontroller and I am not sure I understand why).

Reply:
The resolution happens with a mutating admission webhook, so the Pod never has `mysql`